components/icon/stories/template.js

```diff
@@ -242,42 +166,212 @@ export const Template = ({
 	const classList = {
 		[rootClass]: true,
 		[`spectrum-UIIcon-${iconName}`]: !!(setName === "ui"),
-		[`${rootClass}--${scale}`]: !!(setName === "ui" && scale),
 		[`${rootClass}--size${size?.toUpperCase()}`]: !!(
 			(!setName || setName === "workflow") &&
 			size
 		),
 		...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 	};
 
-	if (svgString) {
-		const classesAsString = Object.entries(classList).reduce((acc, [key, value]) => {
-			if (value) acc += `${key} `;
-			return acc;
-		}, "");
+	/**
+	 * Display full SVG file markup from global IconLoader data, when not using a reference to the sprite sheet.
+	 */
+	if (!useRef) {
+		let svgString;
+		if (loaded?.icons && loaded?.icons[setName]?.[idKey]) {
+			svgString = loaded.icons[setName][idKey];
+		}
+
+		// Return the individual SVG's entire markup.
+		if (svgString) {
+			const classesAsString = Object.entries(classList).reduce((acc, [key, value]) => {
+				if (value) acc += `${key} `;
+				return acc;
+			}, "");
 
-		return html`${unsafeSVG(
-			svgString.replace(/<svg/, `<svg class="${classesAsString}" focusable="false" aria-hidden="true" role="img"`)
-		)}`;
+			return html`${unsafeSVG(
+				svgString.replace(/<svg/, `<svg class="${classesAsString}" focusable="false" aria-hidden="true" role="img"`)
```

Comment:
This doesn't seem to pick-up the `fill` settings. Maybe add a style attribute here to pull in the fill colors?

Reply:
This is fixed.